### ISSUE TITLE
🚀 Feature Request: wrangler init should also generate an example test

### ISSUE THREAD

**Issue by rozenmd** [cloudflare/workers-sdk#1774]
### Describe the solution

TODO:
- [x] Jest for JS
- [x] Vitest for JS
- [ ] Jest for TS
- [ ] Vitest for TS

Just a test that index.ts/index.js returns "Hello World!" as expected

This should probably be an option as part of the init flow, you actually have to do a bunch of stuff to get Jest working:

#### Setting up Jest for TypeScript

```bash
npm i -D jest babel-jest @babel/core @babel/preset-env @babel/preset-typescript @types/jest
```

- add `"test": "jest"` to package.json scripts
- create a `babel.config.js` (if babel.config.js already exists, let the user know to add `"@babel/preset-typescript"` to the presets array):
```js
//babel.config.js
module.exports = {
  presets: [
    ["@babel/preset-env", { targets: { node: "current" } }],
    "@babel/preset-typescript",
  ],
};
```

- add `@types/jest` to the types array in tsconfig.json

- add an `index.test.ts`:
```js
//index.test.ts
import { unstable_dev } from 'wrangler'

describe("worker", () => {
	it("should return Hello World", async () => {
		const worker = await unstable_dev(
			"src/index.ts",
			{},
			{ disableExperimentalWarning: true }
		);
		const resp = await worker.fetch();
		if (resp) {
			const text = await resp.text();
			expect(text).toMatchInlineSnapshot(`"Hello World!"`);
		}
		await worker.stop();
	});
});
```

**Comment by petebacondarwin** [comment]
This is a great idea. I love that it actually tests the initial behaviour, which will break once people start changing the Worker.
We have discussed if Jest is the runner we want to use; probably we might offer a selection?

**Comment by rozenmd** [comment]
Un-assigning myself as I've got plenty of stuff to focus on - feel free to ask questions if you want to pick this up!

**Comment by JacobMGEvans** [comment]
Ooooh Vitest! I like that idea 😄

**Comment by JacobMGEvans** [comment]
> +1 To offering choices between multiple tests suites, would be nice to at least include Ava & Mocha.

Could consider those, since they are also major commonly used runners/suites

**Comment by penalosa** [comment]
We'll address this as part of the upcoming work on Miniflare's unit testing environment cc @admah @mrbbot

**Comment by admah** [comment]
The "Hello World" Worker templates (JS and TS) generated via C3 now include `@cloudflare/vitest-pool-workers` and an example test.

Closed by #5189
